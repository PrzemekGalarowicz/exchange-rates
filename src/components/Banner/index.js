export { default } from './Banner.component.js';
