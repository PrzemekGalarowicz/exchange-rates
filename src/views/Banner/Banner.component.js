import React from 'react';
import PropTypes from 'prop-types';
import { Section, Heading } from 'react-bulma-components';

const Banner = ({ title, subtitle }) => (
  <Section>
    <Heading>{title}</Heading>
    {subtitle && <Heading subtitle>{subtitle}</Heading>}
  </Section>
);

Banner.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string
};

export default Banner;
