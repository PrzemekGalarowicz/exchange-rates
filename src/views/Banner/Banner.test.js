import React from 'react';
import renderer from 'react-test-renderer';
import Banner from './Banner.component';

const defaultProps = {
  title: 'Title',
  subtitle: 'Subtitle'
};

describe('<Banner />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Banner {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly without subtitle', () => {
    const tree = renderer.create(<Banner title={defaultProps.title} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
