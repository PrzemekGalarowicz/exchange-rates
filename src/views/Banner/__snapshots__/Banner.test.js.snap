// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> renders correctly 1`] = `
<section
  className="section"
>
  <h1
    className="title"
  >
    Title
  </h1>
  <h1
    className="subtitle"
  >
    Subtitle
  </h1>
</section>
`;

exports[`<Banner /> renders correctly without subtitle 1`] = `
<section
  className="section"
>
  <h1
    className="title"
  >
    Title
  </h1>
</section>
`;
